### DISCUSSION TITLE
Javascript function is not defined when using @script/@endscript tags

### DISCUSSION THREAD

**Discussion by rooselle** [livewire/livewire#7358] [7. Bugs (archived)]
### Livewire version

3.2.1

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a new Livewire component with following code in the blade template:
```html
<div>
    @script
    <script type="text/javascript">
        function hello() {
            console.log('hello');
        }

        hello();
    </script>
    @endscript
</div>
```
2. Use the component somewhere
3. Display the page where the component is and open the console of your web browser
4. Open the console: `Uncaught ReferenceError: hello is not defined` is displayed
6. Replace the code of the blade template by following code:
```html
<div>
    @script
    <script type="text/javascript">
        let hi = 'hi!';

        function hello() {
            console.log('hello');
        }

        hello();
    </script>
    @endscript
</div>
```
7. "hello" is displayed in the console
8. Alternatively, not using `@script/@endscript` (instead of using a useless var) resolves the problem.

In conclusion: I need to create a useless variable (writing a comment instead doesn't work, not that I would like this to be a solution) so that my function is defined, when using `@script/@endscript` tags. And I need to use them in order to use the `$wire` variable. Do you have any idea why I need to use this workaround?

Thanks for your input! :pray: 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by potsky** [comment]
Same problem for me

**Comment by richcarni** [comment]
I think it's the same/related issue for me...

Without some sort of dummy line before `var input`, I'm getting a: SyntaxError: Unexpected token 'var'

```
@script
<script >
    if(0);
    var input = document.getElementById('address');
    var options;
```

**Comment by F-Yousri** [comment]
facing the same issue with Livewire 3.4.9

**Comment by F-Yousri** [comment]
after multiple trials, it seems like the code in `@script  @enscript` gets executed once the component is loaded. Even code in a function gets executed even without a call to the function. Maybe this is the intended behavior to run some js code every time the component is reloaded.

I wanted to do a backend request after some drag and drop action is done in the component so I needed to use the $wire object. I started like this (which didn't work as expected):

```
@script
<script>
function drop(event, el) {
...
    updateFormation(playerId, parent.id, offsetX, offsetY);
}

function updateFormation(playerId, position, offsetX, offsetY) {
    $wire.updateFormation(playerId, position, offsetX, offsetY);
}
</script>
@endscript
```

but after realizing that code in `@script` runs immediately even if wrapped in function, I removed it, pushed the script into a script stack instead and used the global Livewire variable to get the $wire object.
```
<script>
function drop(event, el) {
...
    updateFormation(playerId, parent.id, offsetX, offsetY);
}

function updateFormation(playerId, position, offsetX, offsetY) {
       $wire = Livewire.getByName('my.component-name')[0];
       $wire.updateFormation(playerId, position, offsetX, offsetY);
}
```

**Reply by lucasweaver10** [reply]
Thank you so much you genius you 🙏

**Comment by joshhanley** [comment]
Creating a function inside `@script` and making it global isn't supported, see @calebporzio's comment here https://github.com/livewire/livewire/discussions/7348#discussioncomment-8233106

But if you assign the function to the window object, then it will be accessible:

```blade
@script
<script>
window.myFunction = () => {}
</script>
@endscript
```

**Comment by dougkusanagi** [comment]
We're in october and this still happens... But even that doesn't work with `onclick="updateCropper()"` in a button.

```blade
@script
    <script>
        let a;    // Without this useless line for a reason the updateCropper function calling cause an error saying it doesn't exists

        function updateCropper() {
            const image = document.getElementById('image');
            const aspect_ratio = document.getElementById('aspectRatio');
            const cropper = new Cropper(image, {
                aspectRatio: aspect_ratio.checked ? 1 : NaN,
                viewMode: 1,
                autoCropArea: 1,
                ready: () => croppable = true,
            });
        }

        updateCropper();
    </script>
@endscript
```
